lib/src/json_schema/validator.dart

```diff
@@ -343,49 +353,76 @@ class Validator {
     }
   }
 
+  void _itemsValidation2020(JsonSchema schema, Instance instance) {
+    final int actual = instance.data.length;
+    final int end = min(schema.prefixItems?.length ?? 0, actual);
+    if (schema.prefixItems != null) {
+      var items = schema.prefixItems;
+      for (int i = 0; i < end; i++) {
+        assert(items[i] != null);
+        final itemInstance = Instance(instance.data[i], path: '${instance.path}/$i');
+        _validate(items[i], itemInstance);
+        _setItemAsEvaluated(i);
+      }
+    }
+
+    if (schema.items != null) {
+      for (int i = end; i < actual; i++) {
+        final itemInstance = Instance(instance.data[i], path: '${instance.path}/$i');
+        _validate(schema.items, itemInstance);
+        _setItemAsEvaluated(i);
+      }
+    }
+  }
+
   void _itemsValidation(JsonSchema schema, Instance instance) {
     final int actual = instance.data.length;
 
-    if (_vocabulary.contains(SupportedVocabularies.APPLICATOR)) {
-      final singleSchema = schema.items;
-      if (singleSchema != null) {
-        instance.data.asMap().forEach((index, item) {
-          final itemInstance = Instance(item, path: '${instance.path}/$index');
-          _validate(singleSchema, itemInstance);
-        });
-        // All the items in this list have been evaluated.
-        _setEvaluatedItemCount(actual);
+    if (_vocabulary.contains(SupportedVocabularies.APPLICATOR_2019) ||
+        _vocabulary.contains(SupportedVocabularies.APPLICATOR_2020)) {
+      if (schema.schemaVersion >= SchemaVersion.draft2020_12) {
+        _itemsValidation2020(schema, instance);
```

Comment:
Instead of making this function bigger, I just make a new smaller function to validate items when validating in draft 2020.